csharp/test/behaviour/connection/transaction/TransactionTest.cs

```diff
@@ -0,0 +1,288 @@
+/*
+ * Copyright (C) 2022 Vaticle
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+using DataTable = Gherkin.Ast.DataTable;
+using System;
+using Xunit;
+using Xunit.Gherkin.Quick;
+
+using com.vaticle.typedb.driver.Test.Behaviour.Connection;
+using com.vaticle.typedb.driver.Test.Behaviour.Connection.Database;
+using com.vaticle.typedb.driver.Test.Behaviour.Connection.Session;
+using com.vaticle.typedb.driver.Test.Behaviour.Connection.Transaction;
+
+namespace com.vaticle.typedb.driver.Test.Behaviour.Connection.Transaction
+{
+    [FeatureFile("external/vaticle_typedb_behaviour/connection/transaction.feature")]
+    public class TransactionTest
+        : Feature
+        , IDisposable
+        , IClassFixture<TransactionSteps>
+        , IClassFixture<SessionSteps>
+        , IClassFixture<DatabaseSteps>
+        , IClassFixture<ConnectionSteps>
+    {
+        public TransactionTest()
+            : base()
+        {
+            _transactionSteps = new TransactionSteps();
+            _sessionSteps = new SessionSteps();
+            _connectionSteps = new ConnectionSteps();
+            _databaseSteps = new DatabaseSteps();
+        }
+
+        public void Dispose()
+        {
+            _connectionSteps.Dispose();
+        }
+
+        [Given(@"typedb starts")]
+        [When(@"typedb starts")]
+        public void TypeDBStarts()
+            => _connectionSteps.TypeDBStarts();
```

Comment:
Do we really have to repeat the steps like this in _each_ test??

Reply:
I haven't deleted the unused ones, it's WIP, but the ones we mention in the feature file -- yes... My previous design helped to avoid it, but since I understood that we have to reuse steps in different tests (e.g. transaction test uses transaction and session steps, while session test uses transaction and session steps AS WELL) and we can't have multiple inheritance, I've decided to stick to [the second option from the library's docs](https://github.com/ttutisani/Xunit.Gherkin.Quick/blob/master/docs/reuse-step-implementation-across-features.md).

I don't think that it's that bad, considering that it's just for one driver and we won't likely double the amount of tests for it, so a bit of copy-paste could be tolerated. Still, I can try to think about something else with my updated understanding of what's going on inside the tests.

Reply:
Will try to refactor using partial classes.

Reply:
Refactored here: https://github.com/vaticle/typedb-driver/commit/e9b2ef79b99cecdaba3d77aa04ebd008c0721381